webapp/package.json

```diff
@@ -13,62 +13,69 @@
-    "@babel/runtime": "7.10.4",
-    "@emotion/babel-preset-css-prop": "10.0.27",
-    "@emotion/core": "10.0.28",
-    "@types/enzyme": "3.10.5",
-    "@types/jest": "26.0.4",
-    "@types/node": "14.0.20",
-    "@types/react": "16.9.41",
-    "@types/react-dom": "16.9.8",
-    "@types/react-intl": "3.0.0",
-    "@types/react-redux": "7.1.9",
+    "@babel/preset-env": "7.16.11",
+    "@babel/preset-react": "7.16.7",
+    "@babel/preset-typescript": "7.16.7",
+    "@babel/runtime": "7.16.7",
+    "@emotion/babel-preset-css-prop": "11.2.0",
+    "@emotion/core": "10.3.1",
+    "@testing-library/jest-dom": "5.16.1",
+    "@types/babel__core": "7.1.18",
+    "@types/babel__template": "7.4.1",
+    "@types/enzyme": "3.10.11",
+    "@types/jest": "27.4.0",
+    "@types/node": "17.0.12",
+    "@types/react": "^16.14.0",
+    "@types/react-dom": "17.0.11",
+    "@types/react-redux": "7.1.22",
     "@types/react-router-dom": "5.1.5",
     "@types/react-transition-group": "4.4.0",
-    "@typescript-eslint/eslint-plugin": "3.6.0",
-    "@typescript-eslint/parser": "3.6.0",
+    "@typescript-eslint/eslint-plugin": "5.10.1",
+    "@typescript-eslint/parser": "5.10.1",
     "babel-eslint": "10.1.0",
-    "babel-jest": "26.1.0",
-    "babel-loader": "8.1.0",
-    "babel-plugin-typescript-to-proptypes": "1.3.2",
-    "css-loader": "3.6.0",
+    "babel-loader": "8.2.2",
+    "babel-plugin-formatjs": "10.3.7",
+    "babel-plugin-typescript-to-proptypes": "2.0.0",
+    "css-loader": "6.5.1",
     "enzyme": "3.11.0",
-    "enzyme-adapter-react-16": "1.15.2",
-    "enzyme-to-json": "3.5.0",
-    "eslint": "7.4.0",
-    "eslint-import-resolver-webpack": "0.12.2",
-    "eslint-plugin-import": "2.22.0",
-    "eslint-plugin-react": "7.20.3",
-    "eslint-plugin-react-hooks": "4.0.6",
-    "file-loader": "6.0.0",
+    "enzyme-adapter-react-16": "1.15.6",
+    "enzyme-to-json": "3.6.2",
+    "eslint": "8.8.0",
+    "eslint-import-resolver-alias": "1.1.2",
+    "eslint-import-resolver-webpack": "0.13.2",
+    "eslint-plugin-babel": "5.3.1",
+    "eslint-plugin-cypress": "2.12.1",
+    "eslint-plugin-header": "3.1.1",
+    "eslint-plugin-import": "2.25.4",
+    "eslint-plugin-mattermost": "github:mattermost/eslint-plugin-mattermost#46ad99355644a719bf32082f472048f526605181",
+    "eslint-plugin-no-only-tests": "2.6.0",
+    "eslint-plugin-react": "7.28.0",
+    "eslint-plugin-react-hooks": "4.3.0",
+    "file-loader": "6.2.0",
     "identity-obj-proxy": "3.0.0",
-    "jest": "26.1.0",
-    "jest-canvas-mock": "2.2.0",
-    "jest-junit": "11.0.1",
-    "mattermost-webapp": "github:mattermost/mattermost-webapp#23f5f93d9f12a7e2b5623e5cee6814366abd9a0f",
-    "node-sass": "4.14.1",
-    "sass-loader": "9.0.2",
-    "style-loader": "1.2.1",
-    "webpack": "4.43.0",
-    "webpack-cli": "3.3.12"
+    "isomorphic-fetch": "3.0.0",
+    "jest": "27.4.7",
+    "jest-canvas-mock": "2.3.1",
+    "jest-junit": "13.0.0",
+    "mattermost-webapp": "github:mattermost/mattermost-webapp",
```

Comment:
In particular, is this going to cause npm to try to pick up new versions of the web app whenever someone does an npm install like it likes to do for non-exact versions of other dependencies?

Reply:
We can definitely do a specific version hash to pull based on the plugin now. I did this because I kept hitting issues downloading the packages and this seemed to be one of the only solutions that worked.

Reply:
Was it the one about the integrity check being wrong? That'd happen for every dependency pulled directly from GitHub unfortunately. If that's the case, we've done some similar fixes [in the web app](https://github.com/mattermost/mattermost-webapp/pull/9617) and [other plugins](https://github.com/mattermost/mattermost-plugin-calls/pull/3).

I think doing that and depending on a specific version would be safer since there's no risk of the web app being accidentally updated by NPM and breaking something if any of the functions used were to change or be removed

Reply:
That's fair. We could essentially tie this dependency to the ESR, if we wanted to leave the `mattermost-webapp` package in here.

Reply:
@coltoneshaw I'm thinking we should import the `mattermost-webapp` repo from a specific git commit and follow @hmhealey's strategy of avoiding the integrity check